src/btree/row_srch.c

```diff
@@ -781,23 +781,28 @@ __wt_row_random_leaf(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt)
 }
 
 /*
- * __wt_row_random_descent --
- *	Find a random leaf page in a row-store tree.
+ * __wt_random_descent --
+ *	Find a random leaf page in a tree.
  */
 int
-__wt_row_random_descent(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt)
+__wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, bool eviction)
```

Comment:
In response to the other question - I'm inclined to keep the single function for now. The function isn't that long, but I can easily imaging bug fixes coming here, and I don't see a way to share the common code paths.